discussions/tasks.py

```diff
@@ -98,18 +99,19 @@ def sync_channel_memberships():
         return
 
     # establish when we'll stop syncing this "batch"
-    end_time = time.time() + SYNC_MEMBERSHIPS_LOCK_TTL_SECONDS
+    end_time = now_in_utc() + timedelta(seconds=SYNC_MEMBERSHIPS_LOCK_TTL_SECONDS)
     lock = _get_sync_memberships_lock()
 
-    def should_sync():
-        """Returns True if we should continue syncing"""
-        return end_time > time.time()  # should only sync if the lock hasn't expired
+    def _get_memberships():
+        """Generator for membership ids to sync"""
+        for membership_id in api.get_membership_ids_needing_sync():
+            # this will stop yielding once our lock has expired
+            if end_time > now_in_utc():
+                yield membership_id
 
     if lock.acquire():
         try:
-            api.sync_channel_memberships(should_sync)
-        except LockError as ex:
-            log.debug("Unable to get lock for syncing channel memberships: %s", ex)
+            api.sync_channel_memberships(_get_memberships())
         finally:
             try:
                 lock.release()
```

Comment:
What case would we not own the lock anymore? When it expires?

Reply:
Correct, if we happen to run past the expiration of the lock and another process recreates the lock it will raise this error.